service/src/test/java/org/sunbird/learner/actors/LearnerStateActorTest.java

```diff
@@ -196,4 +202,119 @@ public void testForGetContentByContentIds() {
     Response res = probe.expectMsgClass(duration("10 second"), Response.class);
     Assert.assertNotNull(res);
   }
+
+  @Test
+  public void testGetCourseByUserIdAndCourseMetadata() {
+
+    TestKit probe = new TestKit(system);
+    ActorRef subject = system.actorOf(props);
+
+    Request request = new Request();
+    Map<String, Object> map = new HashMap<>();
+    map.put(JsonKey.USER_ID, "user1");
+    request.setRequest(map);
+    Map<String, String[]> queryParams = new HashMap<>();
+    String s[] = {JsonKey.END_DATE, JsonKey.STATUS};
+    request.getContext().put(JsonKey.FIELDS, s);
+    request.getContext().put(JsonKey.URL_QUERY_STRING, "String");
+    request.setOperation(ActorOperations.GET_COURSE.getValue());
+    mockEsUtilforUserNcourseBatch();
+    mockContentUtil();
+    subject.tell(request, probe.getRef());
+    Response res = probe.expectMsgClass(duration("10 second"), Response.class);
+    System.out.print("$$$$$$$$$$$$44 " + res.getResult());
+    ProjectLogger.log(
+        "Result After Completeing test is " + res.getResult() + " ", LoggerEnum.INFO.name());
+    Assert.assertNotNull(res);
+  }
+
+  @Test
+  public void testGetCourseByUserIdAndCourseMetadataInvalidvalues() {
+
+    TestKit probe = new TestKit(system);
```

Comment:
make the two new introduced test-case common - introduce testScenario funciton and move all common function to it.